akeneo-design-system/src/components/Table/TableHeaderCell/TableHeaderCell.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {ReactNode, Ref} from 'react';
+import styled, {css} from 'styled-components';
+import {AkeneoThemedProps, getColor} from '../../../theme';
+import {ArrowDownIcon, ArrowUpIcon} from '../../../icons';
+
+type TableSortDirection = 'descending' | 'ascending' | 'none';
+
+type TableHeaderCellProps = {
+  /**
+   * Define if the header can be sorted
+   */
+  isSortable?: boolean;
+
+  /**
+   * Function called when the user click on sort icon, required when isSortable
+   */
+  onDirectionChange?: (direction: TableSortDirection) => void;
+
+  /**
+   * Define the sort direction
+   */
+  sortDirection?: TableSortDirection;
+
+  /**
+   * Content of the header cell
+   */
+  children?: ReactNode;
+};
+
+const HeaderCellContainer = styled.th<{isSortable: boolean; isSorted: boolean} & AkeneoThemedProps>`
+  background: linear-gradient(to top, #67768a 1px, white 0px);
+  height: 44px;
+  text-align: left;
+  color: ${({isSorted}) => (isSorted ? getColor('purple', 100) : getColor('grey', 100))};
+  font-weight: normal;
+
+  ${({isSortable}) =>
+    isSortable &&
+    css`
+      cursor: pointer;
+    `};
+`;
+
+const HeaderCellContentContainer = styled.span`
+  color: ${getColor('grey', 140)};
+  padding: 0 10px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  + svg {
+    vertical-align: middle;
+  }
+`;
+const TableHeaderCell = React.forwardRef<HTMLTableHeaderCellElement, TableHeaderCellProps>(
+  (
+    {isSortable = false, onDirectionChange, sortDirection = 'none', children, ...rest}: TableHeaderCellProps,
+    forwardedRef: Ref<HTMLTableHeaderCellElement>
+  ) => {
+    if (isSortable && onDirectionChange === undefined) {
```

Comment:
?
```suggestion
    if (isSortable && (onDirectionChange === undefined || sortDirection === null)) {
```
The only (edge) case I see this can happen is if the user creates his cell as following:
<Table.HeaderCell isSortable={true} onDirectionChange={() => {}} sortDirection=null />

Is this something that can happen ?